### DISCUSSION TITLE
Alternative ways to add empty attributes to links?

### DISCUSSION THREAD

**Discussion by PlasmaHH** [squidfunk/mkdocs-material#7359] [Q&A]
I have been trying to get downloading links to work, or better to fix it, since a lot of our files is converted via pandoc. There it automatically sets `{:download}`  for those links which almost works, but the resulting html is `download="download"` which causes the file that is to be downloaded to be called `download.<ext>`  which is certainly not desired.

I figured out so far that it works by specifying `{download=""}` but given that pandoc converts it that way, before I change thousands of occurences I wanted to ask if I have overlooked something and there is a tiny bit more concise syntax or a setting to make `{:download}`  work.

**Comment by squidfunk** [comment]
I don't think there's another way than using `download=""`. However, you could also write an `on_page_markdown` hook to post-process the Markdown and remove the `=""` – should be quite simple. It might also be a good idea to take this to the maintainers of Python Markdown, as this is the project that provides the Attribute Lists implementation.

**Reply by kamilkrzyskow** [reply]
There is also the option to monkey-patch the inner Markdown implementation using the hook, instead of processing whole text files:
- https://github.com/Python-Markdown/markdown/blob/33359faa385f59b84cd87df5f4b0996055a482e2/markdown/extensions/attr_list.py#L162-L185
